### PR TITLE
TRANS-5724 Create pocket data

### PR DESCRIPTION
**Pull request by SjaakBall**
Done:
Ingest random number up to 3 pockets per user through client-api.

Still open:
Ingest Transactions with Transaction.ArrangementId(external Arrangement id) as parent ArrangementId and Transaction.Reference as pocket arrangement Id.
See https://backbase.atlassian.net/browse/NVCTS-404

### CODE REVIEW

File: src/main/java/com/backbase/ct/bbfuel/client/pfm/PocketsRestClient.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        setVersion(SERVICE_VERSION);
+    }
+
+    /**
+     * Ingest Pockets.
+     *
+     * @param pocketPostRequest pocketPostRequest
+     * @return Response
+     */
+    public Response ingestPockets(PocketPostRequest pocketPostRequest) {
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/client/pfm/PocketsRestClient.java:36]: ingestPockets -> ingestPocket

File: src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PocketsConfigurator {
+
+    private static final GlobalProperties globalProperties = GlobalProperties.getInstance();
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java:28]: globalProperties -> GLOBAL_PROPERTIES

File: src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        List<PocketPostRequest> pockets = Collections
+            .synchronizedList(new ArrayList<>());
+
+        int randomAmount = CommonHelpers
+            .generateRandomNumberInRange(globalProperties.getInt(CommonConstants.PROPERTY_POCKETS_MIN),
+                globalProperties.getInt(CommonConstants.PROPERTY_POCKETS_MAX));
+
+        if (isRetail) {
+            log.debug("Generating pockets data from json file.");
+            IntStream.range(0, randomAmount).parallel()
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java:51]: Please avoid `parallel` streams.

File: src/main/java/com/backbase/ct/bbfuel/data/PocketsDataGenerator.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     * Generate PocketPostRequest.
+     * @return PocketPostRequest
+     */
+    public static PocketPostRequest generatePocketPostRequest() {
+
+        BigDecimal amount = CommonHelpers.generateRandomAmountInRange(100L, 9999L);
+        String currency = GLOBAL_PROPERTIES.getString(POCKETS_CURRENCY);
+
+        return new PocketPostRequest()
+            .name(faker.lorem().characters(80))
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/data/PocketsDataGenerator.java:33]: 80 characters for name and icon seem excessive:
`ksjlakjsdlkajlkcn salkdjalksdjlaksjd  laksjdalksjd lkasjd  lkasawidjowidjoqwidj`

Btw, we could use city names and other fake data instead of `lorem`, that will make more sense, have a look at the available options: https://github.com/DiUS/java-faker#fakers

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java [hunk 5]
```diff
@@ -254,6 +259,13 @@ private void ingestTransactions(List<ArrangementId> arrangementIds, boolean isRe
         }
     }
 
+    private void ingestPockets(List<ArrangementId> arrangementIds, boolean isRetail) {
+        if (this.globalProperties.getBoolean(PROPERTY_INGEST_POCKETS)) {
+            arrangementIds.parallelStream()
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:264]: Please avoid `parallel` streams.

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java [hunk 6]
```diff
@@ -254,6 +259,13 @@ private void ingestTransactions(List<ArrangementId> arrangementIds, boolean isRe
...
+                .forEach(arrangementId -> this.pocketsConfigurator.ingestPockets(arrangementId, isRetail));
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:265]: I think we should ingest pockets for only one of the user's arrangements, we should create the parent arrangement and then ingest pockets for it.

File: src/main/resources/data/retail/pockets.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      "amountCurrency": {
+        "amount": "800.00",
+        "currencyCode": "EUR"
+      },
+      "deadline": "2021-08-14"
+    }
+  },
+  {
+    "name": "Unlimited Happiness",
+    "icon": "medicine",
```
Comment (lehphyro) [inline src/main/resources/data/retail/pockets.json:26]: I didn't know medicine = unlimited happiness :p

File: src/main/resources/data/retail/pockets.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
+[
+  {
+    "name": "Travel Europe",
+    "icon": "travel",
```
Comment (lehphyro) [inline src/main/resources/data/retail/pockets.json:4]: Did you check if these icon names will be the ones configured by default in the frontend?

File: pom.xml [hunk 9]
```diff
@@ -374,6 +385,18 @@
...
                     </execution>
+
+                    <execution>
+                        <id>download-pocket-tailor-client-api-spec</id>
+                        <goals>
+                            <goal>download-single</goal>
+                        </goals>
+                        <phase>generate-sources</phase>
+                        <configuration>
+                            <url>https://artifacts.backbase.com/specs/pocket-tailor/</url>
```
Comment (vinibackbase) [inline pom.xml:396]: BB-Fuel should only contain dependencies that are released to repo. In order to point to specs on artifacts, this needs to be marked a pre-release PR.

File: src/main/resources/data/products.json [hunk 10]
```diff
@@ -47,5 +47,19 @@
...
+    "productKindName": "Pocket Parent",
+    "productTypeId": "8888888",
+    "productTypeName": "Pocket Parent"
+  },
+  {
+    "id": "9",
+    "productKindId": "kind9",
+    "productKindName": "Pocket",
+    "productTypeId": "9999999",
+    "productTypeName": "Pocket"
```
Comment (vinibackbase) [inline src/main/resources/data/products.json:63]: Are these product kinds already configured within product summary? Because otherwise they might be categorised under "Custom Product Kinds". Not sure if that's an issue for pockets.

File: src/main/resources/data/retail/pockets.json [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    "icon": "medicine",
+    "goal": {
+      "amountCurrency": {
+        "amount": "200000.00",
+        "currencyCode": "EUR"
+      },
+      "deadline": "2035-02-28"
+    }
+  }
+]
```
Comment (vinibackbase) [inline src/main/resources/data/retail/pockets.json:35]: Please add the newline at the end of the file.

File: src/main/resources/data/job-profiles.json [hunk 12]
```diff
@@ -16,6 +16,16 @@
       "retail"
     ],
     "permissions": [
+      {
+        "businessFunction": "Manage Pockets",
```
Comment (vinibackbase) [inline src/main/resources/data/job-profiles.json:20]: what kind of users should have this permission? because we have the retail-banking specific job profile json file as well.

File: src/main/resources/data.properties [hunk 13]
```diff
@@ -55,6 +55,11 @@ actions.max=2
 # Ingest billpay
 ingest.billpay=false
 ingest.billpay.accounts=false
+# ingest pockets
+ingest.pockets=false
+pockets.min=1
+pockets.max=3
+pockets.currency=EUR
```
Comment (vinibackbase) [inline src/main/resources/data.properties:62]: please also add a property specifying the path to the pockets.json file instead of hardcoding it on the code.

File: src/main/java/com/backbase/ct/bbfuel/input/PocketsReader.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Arrays;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PocketsReader extends BaseReader {
+
+    public PocketPostRequest loadSingle() {
+        return CommonHelpers
+            .getRandomFromList(load(globalProperties.getString(CommonConstants.PROPERTY_POCKETS_DATA_JSON)));
```
Comment (vinibackbase) [inline src/main/java/com/backbase/ct/bbfuel/input/PocketsReader.java:17]: please also add a property specifying the path to the pockets.json on data.properties file instead of hardcoding it on the code. So we can have multiple files in the case we need to test edge cases in the future.

**Review by vinibackbase — CHANGES_REQUESTED** [review]

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java (hunk 6 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:265]: How do I create the parent arrangement? And that parent arrangement should ingest the pockets?

Comment (lehphyro) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:265]: Check how this method is used: https://github.com/Backbase/bb-fuel/blob/master/src/main/java/com/backbase/ct/bbfuel/client/productsummary/ArrangementsIntegrationRestClient.java#L37

File: pom.xml (hunk 9 above)
Comment (SjaakBall) [reply pom.xml:396]: I will indeed mark it with a todo as I already did for the pocket-tailor dependencies.

File: src/main/java/com/backbase/ct/bbfuel/client/pfm/PocketsRestClient.java (hunk 1 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/client/pfm/PocketsRestClient.java:36]: I will change it

File: src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java (hunk 2 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java:28]: I will change it

File: src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java (hunk 3 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java:51]: I will remove the parallel method

File: src/main/java/com/backbase/ct/bbfuel/data/PocketsDataGenerator.java (hunk 4 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/data/PocketsDataGenerator.java:33]: I will use some fancy fake data that makes more sense

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java (hunk 5 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:264]: I will remove the parallel method

File: src/main/resources/data/retail/pockets.json (hunk 8 above)
Comment (SjaakBall) [reply src/main/resources/data/retail/pockets.json:4]: I am trying to get the default list...

File: src/main/resources/data/retail/pockets.json (hunk 11 above)
Comment (SjaakBall) [reply src/main/resources/data/retail/pockets.json:35]: I will do that

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java (hunk 6 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:265]: I need some more instructions to be able to create a parent arrangement. Should it be in the ingestPocket method or before it is called? What I use as input data?

File: src/main/resources/data/products.json (hunk 10 above)
Comment (SjaakBall) [reply src/main/resources/data/products.json:63]: @lehphyro Do you know if these product kinds are already configured within product summary?

File: src/main/resources/data/job-profiles.json (hunk 12 above)
Comment (SjaakBall) [reply src/main/resources/data/job-profiles.json:20]: @lehphyro Could you answer which user have this permission?

File: src/main/resources/data.properties (hunk 13 above)
Comment (SjaakBall) [reply src/main/resources/data.properties:62]: I will do that

File: src/main/java/com/backbase/ct/bbfuel/input/PocketsReader.java (hunk 14 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/input/PocketsReader.java:17]: I will use a property

File: src/main/resources/data/products.json (hunk 10 above)
Comment (lehphyro) [reply src/main/resources/data/products.json:63]: You know that because you use them to create pockets :)
Here is a PR created by yourself: https://stash.backbase.com/projects/PFM/repos/pocket-management/pull-requests/14/diff#pocket-tailor/src/main/resources/application.yml

File: src/main/resources/data/job-profiles.json (hunk 12 above)
Comment (lehphyro) [reply src/main/resources/data/job-profiles.json:20]: Every retail banking user needs to be able to use Pockets.

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java (hunk 6 above)
Comment (lehphyro) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:265]: You need to make this call work: https://stash.backbase.com/projects/PFM/repos/pocket-management/browse/pocket-tailor/src/main/java/com/backbase/dbs/pfm/pockets/service/impl/PocketTailorServiceImpl.java#62

There is a mock service that always returns the same arrangement when it's asked to created a parent one: https://stash.backbase.com/projects/SDLC/repos/beck-helm/browse/helmfile.yaml#2227

File: src/main/resources/data/job-profiles.json (hunk 12 above)
Comment (SjaakBall) [reply src/main/resources/data/job-profiles.json:20]: I will do that.

File: src/main/resources/data/products.json (hunk 10 above)
Comment (SjaakBall) [reply src/main/resources/data/products.json:63]: Annatolli Volkodav provided me this info [https://stash.backbase.com/projects/AR/repos/arrangement-manager/pull-requests/393/diff#persistence/src/main/resources/sql/mssql/create/arrangement-manager.sql?t=128] , which seems to correspond with the settings in the products.json file.

**Comment by vinibackbase** [comment]
Once you change the PR to a pre-release one, we can merge

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java (hunk 6 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:272]: Attempt to implement pocket parent arrangement, but config of PocketsMockArrangementRestClient results in ERROR: `nodename nor servname provided, or not known`
I do not know how to solve this @lehphyro .

Comment (lehphyro) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:272]: That means the service you're trying to call doesn't exist.

File: src/main/resources/application-k8s-beta.yml [hunk 15]
```diff
@@ -19,6 +19,8 @@ bb-fuel:
     transactions: http://transactions.${environment.name}.${environment.domain}/integration-api
     user: http://user-manager.${environment.name}.${environment.domain}/integration-api
     accountStatement: http://accountstatementsintegration.${environment.name}.${environment.domain}/service-api
+    pockets: http://edge.${environment.name}.${environment.domain}/api/pocket-tailor/client-api
+    pocketsArrangements: http://pocketsprovidermock.${environment.name}.${environment.domain}/service-api
```
Comment (bb-pratishtha) [inline src/main/resources/application-k8s-beta.yml:23]: Here, you are trying to call arrangement-outbound integration service which is deployed as https://stash.backbase.com/projects/SDLC/repos/beck-helm/browse/helmfile.yaml#2221

File: src/main/resources/application-baas.yml [hunk 16]
```diff
@@ -16,6 +16,8 @@ bb-fuel:
     transactions: https://app.${environment.name}.${environment.domain}/transactions/integration-api
     user: https://app.${environment.name}.${environment.domain}/user-manager/integration-api
     accountStatement: https://app.${environment.name}.${environment.domain}/accountstatementsintegration/service-api
+    pockets: http://app.${environment.name}.${environment.domain}/api/pocket-tailor/client-api
+    pocketsArrangements: http://app.${environment.name}.${environment.domain}/pocketsprovidermock/service-api
```
Comment (bb-pratishtha) [inline src/main/resources/application-baas.yml]: we need to check the baas deployment and add the appropriate mock service here and add it in baas deployment if doesn't exist there.

File: src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java (hunk 6 above)
Comment (bb-pratishtha) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:272]: PocketsMockArrangementRestClient , you are trying to use arrangement outbound mock service which is deployed as follows 
https://stash.backbase.com/projects/SDLC/repos/beck-helm/browse/helmfile.yaml#2221. please change pocketsArrangements: http://app.${environment.name}.${environment.domain}/pocketsprovidermock/service-api
 
in src/main/resources/application-k8s-beta.yml

Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:276]: I used the host config value from helm file...

File: src/main/resources/application-baas.yml (hunk 16 above)
Comment (SjaakBall) [reply src/main/resources/application-baas.yml]: I will add a TODO

File: src/main/resources/data/retail/pockets.json (hunk 8 above)
Comment (SjaakBall) [reply src/main/resources/data/retail/pockets.json]: I used this list from Joao: default list of names/images [https://share.goabstract.com/cd3c6f04-5c35-403a-8941-018f71f14f9c]

**Review by vinibackbase — APPROVED** [review]

File: src/main/java/com/backbase/ct/bbfuel/service/AccessGroupService.java [hunk 17]
```diff
@@ -28,6 +28,8 @@
 @RequiredArgsConstructor
 public class AccessGroupService {
 
+    public static final String DATAGROUP_NAME_RETAIL_POCKETS = "Retail Pocket";
```
Comment (lehphyro) [inline src/main/java/com/backbase/ct/bbfuel/service/AccessGroupService.java:31]: DATAGROUP_NAME_RETAIL_POCKETS -> DATA_GROUP_NAME_RETAIL_POCKETS

File: pom.xml [hunk 18]
```diff
@@ -375,6 +375,30 @@
...
+                        </goals>
+                        <phase>generate-sources</phase>
+                        <configuration>
+                            <url>https://repo.backbase.com/specs/pocket-tailor</url>
+                            <fromFile>pocket-tailor-client-api-v2.0.0.yaml</fromFile>
+                        </configuration>
+                    </execution>
+
+                    <execution>
+                        <id>download-arrangement-outbound-origination-mock</id>
```
[the remark this answers is not in the document]
Comment (SjaakBall) [reply pom.xml]: changed

File: pom.xml [hunk 19]
```diff
@@ -396,16 +420,14 @@
                             <url>https://repo.backbase.com/specs/arrangement-manager</url>
                             <fromFile>arrangement-integration-inbound-api-v2.2.0.yaml</fromFile>
                         </configuration>
-                    </execution>
-                </executions>
+                    </execution>                </executions>
```
[the remark this answers is not in the document]
Comment (SjaakBall) [reply pom.xml]: corrected

File: src/main/java/com/backbase/ct/bbfuel/client/productsummary/ArrangementsIntegrationRestClient.java [hunk 20]
```diff
@@ -47,6 +47,35 @@ public ArrangementsPostResponseBody ingestArrangement(ArrangementsPostRequestBod
...
+     * @param arrangement the parent pocket arrangement
+     * @return ArrangementsPostResponseBody
+     */
+    public ArrangementsPostResponseBody ingestParentPocketArrangementAndLogResponse(ArrangementsPostRequestBody arrangement) {
+        ArrangementsPostResponseBody arrangementsPostResponseBody = null;
+        Response response = ingestParentPocketArrangement(arrangement);
+        if (isBadRequestExceptionWithErrorKey(response, "arrangements.api.alreadyExists.arrangement")) {
+            log.info("Arrangement [{}] already exists, skipped ingesting this arrangement", arrangement.getProductId());
+        } else if (response.statusCode() != SC_CREATED) {
+            log.info("Arrangement [{}] already ingested", arrangement.getProductId());
```
[the remark this answers is not in the document]
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/client/productsummary/ArrangementsIntegrationRestClient.java]: removed else statement

File: src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+     */
+    public String ingestPocketParentArrangementAndSetEntitlements(LegalEntityBase legalEntity) {
+        // -> creating parent pocket arrangement for legal entity
+        log.debug("Going to ingest a parent pocket arrangement for external legal entity ID: [{}]", legalEntity);
+
+        String parentPocketArrangementId = null;
+        ArrangementsPostResponseBody arrangementsPostResponseBody = ingestParentPocketArrangement(legalEntity);
+
+        if (arrangementsPostResponseBody != null
+            && StringUtils.isNotEmpty(arrangementsPostResponseBody.getId())) {
```
[the remark this answers is not in the document]
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java]: removed

File: src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+
+        if (parentPocketArrangementId != null) {
+            // -> Now setting entitlements by dataGroup (functionGroup is already managed by setting
+            //      permissions in retail/job-profiles.json: jobProfileName: 'Retail User' )
+            // -> Updating dataGroup makes the method accessControlClient.verifyCreateAccessToArrangement(arrangementId)
+            //      in PocketTailorServiceImpl succeed, by returning all relevant arrangements with
+            //      usersApi.getArrangementPrivileges
+            // -> Updating functionGroup makes @PreAuthorize("checkPermission... in PocketTailorServiceImpl work, by
+            //      AccessControlValidatorImpl.checkPermissions succeed
+            updateDataGroupForPockets(parentPocketArrangementId, legalEntity);
```
[the remark this answers is not in the document]
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/configurator/PocketsConfigurator.java:59]: changed

File: src/main/java/com/backbase/ct/bbfuel/input/PocketsReader.java [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+
+        try {
+            //TODO TRANS-5724 parsing deadline generates parsing error
+            PocketPostRequest[] parsedPockets = ParserUtil.convertJsonToObject(uri, PocketPostRequest[].class);
+            pockets = Arrays.asList(parsedPockets);
+        } catch (IOException e) {
+            log.error("Failed parsing file with Pockets", e);
+            throw new InvalidInputException(e.getMessage(), e);
+        }
+        return pockets;
```
[the remark this answers is not in the document]
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/input/PocketsReader.java]: done

File: src/main/java/com/backbase/ct/bbfuel/service/AccessGroupService.java (hunk 17 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/service/AccessGroupService.java]: renamed

**Review by lehphyro — APPROVED** [review]

File: src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java [hunk 24]
```diff
@@ -59,15 +66,19 @@
...
     private final ActionsConfigurator actionsConfigurator;
     private final BillPayConfigurator billpayConfigurator;
+    private final PocketsConfigurator pocketsConfigurator;
     private final LegalEntityService legalEntityService;
     private final AccountStatementsConfigurator accountStatementsConfigurator;
     private final PositivePayConfigurator positivePayConfigurator;
+    private final UserPresentationRestClient userPresentationRestClient;
+    private final PocketTailorActuatorClient pocketTailorActuatorClient;
 
     /**
```
Comment (richkum92) [inline src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java:76]: You can include our service name in the comment

**Review by richkum92 — COMMENTED** [review]
I checked out your branch and did mvn clean install. Build failed as there is 1 test failure.
But Pocket is getting ingested when I run the bbfuelapplication with VM options mentioned by you in US.

File: src/main/resources/data/retail/pockets.json [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+    "icon": "entertainment",
+    "goal": {
+      "amountCurrency": {
+        "amount": "800.00",
+        "currencyCode": "EUR"
+      },
+      "deadline": "2036-08-05"
+    }
+  },
+  {
```
Comment (richkum92) [inline src/main/resources/data/retail/pockets.json:45]: It would be nice to give one Pocket example without goal

**Review by richkum92 — APPROVED** [review]

File: src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java (hunk 24 above)
Comment (SjaakBall) [reply src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java:76]: Done

File: src/main/resources/data/retail/pockets.json (hunk 25 above)
Comment (SjaakBall) [reply src/main/resources/data/retail/pockets.json:45]: Changed it and also fixed the failing unit test you mentioned above
